Aagam_Shah/postgreAssignment/src/controller/marks.controller.ts

```diff
@@ -0,0 +1,64 @@
+const db = require('../config/database').pool;
+import { Context } from "vm";
+import * as validation from '../helper/validation';
+
+exports.marksOfStudent = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(validation.isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * from school.marks where studentid = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.highestMarkOfSubject = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(validation.isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * from school.marks 
+        where subjectid = $1
+        order by marks desc
+        limit 1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
```

Comment:
there a pattern her and code is repeating can we write something generic and handle error at on place. wrap every thing in try catch block and form validation throw error with code and message and from service also where we do db cal throw error code and error message. same can be used in catch block. let discuss on this if you have doubt